src/main/java/com/perimeterx/models/PXContext.java

```
@@ -385,6 +392,12 @@ public void setBlockAction(String blockAction) {
         }
     }
 
+    private void generateLoginData(HttpServletRequest request, PXConfiguration pxConfiguration) throws PXException {
+        final LoginData loginData = new LoginData(request, pxConfiguration);
+
+        this.setLoginData(loginData);
```

Just confirming that if for whatever reason we aren't on a CI route or there are missing/weird credentials that cause issues, this won't throw an exception that impacts the rest of the flow. Right now I see this may cause an exception in the PXContext constructor right? This isn't an issue?

---

I just maintained the previous logic but after you mentioned it,I've checked in other enforcers and it is fail silently.
Fixed.